apps/wallet/tests/sites-to-cs-messaging.spec.ts

```diff
@@ -0,0 +1,180 @@
+            },
+            false,
+        ],
+        [
+            'UI create wallet',
+            {
+                type: 'keyring',
+                method: 'create',
+                args: {},
+            },
+            false,
+        ],
+    ] as const;
+    for (const [aLabel, aPayload, result] of allTests) {
+        test(aLabel, async ({ context, demoPageUrl }) => {
+            const page = await context.newPage();
+            await page.goto(demoPageUrl);
+            const nextMessage = getInAppMessage(page, aLabel);
+            await page.evaluate(
+                ({ aPayload: payload, aLabel: label }) => {
+                    window.postMessage({
+                        target: 'sui_content-script',
+                        payload: {
+                            id: label,
+                            payload,
+                        },
+                    });
+                },
+                { aPayload, aLabel }
+            );
+            if (result) {
+                expect(await nextMessage).toMatchObject(result);
+            } else {
+                await expect(nextMessage).rejects.toThrow();
+            }
+        });
+    }
+});
+
+test.describe('Wallet interface', () => {
+    let demoPage: Page;
+
+    test.beforeEach(async ({ context, demoPageUrl }) => {
+        demoPage = await context.newPage();
+        await demoPage.goto(demoPageUrl);
+    });
+    test.describe('when not connected', () => {
+        // eslint-disable-next-line no-empty-pattern
+        test('no account is connected', async ({}) => {
+            expect((await demoPage.locator('.account').all()).length).toBe(0);
+            await expect(
+                demoPage.getByRole('button', { name: 'Connect' })
+            ).toBeVisible();
+        });
+        test('executing a transaction fails', async () => {
+            await demoPage
+                .getByRole('button', { name: 'Send transaction' })
+                .click();
+            await expect(demoPage.getByText('Error')).toBeVisible();
+        });
+        test('signing a transaction fails', async () => {
+            await demoPage
+                .getByRole('button', { name: 'Sign transaction' })
+                .click();
+            await expect(demoPage.getByText('Error')).toBeVisible();
+        });
+        test('signing a message', async () => {
+            await demoPage
+                .getByRole('button', { name: 'Sign message' })
+                .click();
+            await expect(demoPage.getByText('Error')).toBeVisible();
+        });
+    });
+    test.describe('when connected', () => {
+        test.beforeEach(async ({ context, demoPageUrl }) => {
+            await demoDappConnect(demoPage, demoPageUrl, context);
+        });
+        test('executing transaction works', async ({ context }) => {
+            const newPage = context.waitForEvent('page');
+            await demoPage.waitForSelector('.account');
```

Comment:
nit: could we apply a `data-testid` to the unordered list element and use that instead? coupling the class name of the list item element to the test code here feels a little brittle

Reply:
Seemed easier since I don't think we ever going to do any styling for the demo app but you are right using data-testid is better